pyat/at/plot/resonances.py

```diff
@@ -0,0 +1,420 @@
+    # only plot normal or skew
+    normalskew = [0, 1]
+    onlyns = kwargs.pop("only", False)
+    if onlyns == "normal":
+        normalskew = [0]
+    if onlyns == "skew":
+        normalskew = [1]
+
+    # orders could be a single int
+    if isinstance(orders, int):
+        orders = [orders]
+    # check that all are larger than 0
+    if sum(n < 1 for n in orders):
+        raise ValueError("Negative resonances are not allowed.")
+
+    if periodicity < 1:
+        raise ValueError("Period must be a positive integer.")
+    verboseprint(f"The periodicity is {periodicity}")
+
+    verboseprint(f"The window is {window}")
+    # check the window
+    windowa = numpy.array(window)
+    if windowa[0] == windowa[1]:
+        raise ValueError("horizontal coordinates must be different")
+    if windowa[2] == windowa[3]:
+        raise ValueError("vertical coordinates must be different")
+    if windowa[1] < windowa[0]:
+        windowa[0], windowa[1] = windowa[1], windowa[0]
+        warnings.warn("Swapping horizontal coordinates", stacklevel=1)
+    if windowa[3] < windowa[2]:
+        windowa[2], windowa[3] = windowa[3], windowa[2]
+        warnings.warn("Swapping vertical coordinates", stacklevel=1)
+    # get xlimits and ylimits
+    the_axeslims = windowa.reshape((2, 2))
+
+    maxreson2calc = numpy.max(orders)
+    verboseprint(f"Farey max order={maxreson2calc}")
+
+    # get the Farey collection, i.e., a list of farey sequences, one per order
+    fareycollectionfloat = {}
+    fareycollectionfrac = {}
+    for nthorder in range(1, maxreson2calc + 1):
+        farey, fracfarey = farey_sequence(nthorder)
+        fareycollectionfloat[nthorder] = farey
+        fareycollectionfrac[nthorder] = fracfarey
+    verboseprint(f"the Farey collection is {fareycollectionfloat}")
+
+    # plot configuration
+    if axes is None:
+        fig = plt.figure()
+        axes = fig.add_subplot(111)
+    else:
+        verboseprint("Axes already exist, ignore window")
+        the_axeslims = numpy.array([axes.get_xlim(), axes.get_ylim()])
+    # min/max to plot lines with slopes
+    minx = numpy.floor(the_axeslims[0, 0])
+    minx = minx - periodicity - numpy.mod(minx, periodicity)
+    maxx = numpy.ceil(the_axeslims[0, 1])
+    maxx = maxx + periodicity - numpy.mod(maxx, periodicity)
+    minmaxxdist = maxx - minx
+    miny = numpy.floor(the_axeslims[1, 0])
+    miny = miny - periodicity - numpy.mod(miny, periodicity)
+    maxy = numpy.ceil(the_axeslims[1, 1])
+    maxy = maxy + periodicity - numpy.mod(maxy, periodicity)
+    minmaxydist = maxy - miny
+
+    linestyle = kwargs.pop("linestyle", None)
+    linecolor = kwargs.pop("linecolor", None)
+    linewidth = kwargs.pop("linewidth", None)
+    addtolabel = kwargs.pop("addtolabel", None)
+    defaultlprop = create_linepalette(
+        linestyle=linestyle,
+        linecolor=linecolor,
+        linewidth=linewidth,
+        addtolabel=addtolabel,
+    )
+    lprop = kwargs.pop("linedict", defaultlprop)
```

Comment:
A suggestion (forget it if you find it irrelevant):
```python
    lprop = create_linepalette(
         linestyle=linestyle,
         linecolor=linecolor,
         linewidth=linewidth,
         addtolabel=addtolabel,
     )
    userprop = kwargs.pop("linedict", {})
    lprop["normal"].update(userprop.get("normal", {}))
    lprop["skew"].update(userprop.get("skew", {}))
```
The present behavior is kept, but it allows the user to overwrite only the lines he wants to change, others are unchanged.

Reply:
Done.